pkg/logagent/logsagent.go

```diff
@@ -66,4 +67,9 @@ func Start() {
 
 	s := tailer.New(config.GetLogsSources(), filePipelinesEntryChannels, a)
 	s.Start()
+
+	// fixme: this is not great, we share the same pipelines as tailers.
```

Comment:
Eventually we should have a ring structure with a `take` method that safely wraps around when the end is reached. This will make the code clean and simple.